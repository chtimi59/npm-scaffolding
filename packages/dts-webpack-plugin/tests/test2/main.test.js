const childProcess = require('node-child-process-extension')
const fs = require('node-fs-extension')
const path = require('path')
const outDir = path.resolve(__dirname, "dist")

describe('dts-webpack-plugin', function () {
    it('test2', async function () {
        await childProcess.extras.exe(`webpack`, __dirname)
        const t1 = await fs.extras.exists(path.resolve(outDir, "types.d.ts"), "file")
        const t2 = await fs.extras.exists(path.resolve(outDir, "lib", "a.d.ts"), "file")
        const t3 = await fs.extras.exists(path.resolve(outDir, "stuff", "baz.d.ts"), "file")
        expect(t1 && t2 && t3).toEqual(true)
        const ret = fs.readFileSync(path.resolve(outDir, "types.d.ts")).toString()
        expect(ret).toMatchSnapshot()
    }, 15000)
})
