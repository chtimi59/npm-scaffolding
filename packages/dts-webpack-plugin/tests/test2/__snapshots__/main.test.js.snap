// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dts-webpack-plugin test2 1`] = `
"
export declare var SOMETHING: string

export declare function bar(v: string): string;
"
`;
